state/protocol/protocol_state/kvstore/models.go

```diff
@@ -158,20 +163,33 @@ func (model *Modelv1) ID() flow.Identifier {
 	return makeVersionedModelID(model)
 }
 
-// Replicate instantiates a Protocol State Snapshot of the given `protocolVersion`.
-// It clones existing snapshot if `protocolVersion = 1`, other versions are not supported yet.
+// Replicate instantiates a Protocol State Snapshot of the given protocolVersion.
+// It clones existing snapshot if protocolVersion = currentVersion.
+// It transitions to next version if protocolVersion = currentVersion+1.
 // Expected errors during normal operations:
 //   - ErrIncompatibleVersionChange if replicating the Parent Snapshot into a Snapshot
 //     with the specified `protocolVersion` is not supported.
 func (model *Modelv1) Replicate(protocolVersion uint64) (protocol_state.KVStoreMutator, error) {
-	version := model.GetProtocolStateVersion()
-	if version == protocolVersion {
+	currentVersion := model.GetProtocolStateVersion()
+	if currentVersion == protocolVersion {
 		// no need for migration, return a complete copy
 		return clone.Clone(model), nil
-	} else {
-		return nil, fmt.Errorf("unsupported replication version %d: %w",
-			protocolVersion, ErrIncompatibleVersionChange)
 	}
+	nextVersion := currentVersion + 1
+	if protocolVersion != nextVersion {
+		// can only Replicate into model with numerically consecutive currentVersion
+		return nil, fmt.Errorf("unsupported replication currentVersion %d, expect %d: %w",
+			protocolVersion, 1, ErrIncompatibleVersionChange)
+	}
+
+	// perform actual replication to the next currentVersion
```

Comment:
Not sure I understand your concern...

The current implementation does:
```go
currentVersion := model.GetProtocolStateVersion()
// ...
nextVersion := currentVersion + 1
if protocolVersion != nextVersion { /* error */ }
```

It enforces that `protocolVersion` is exactly equal to `currentVersion+1`, and `currentVersion` is a property of `ModelV1` (the struct we are defining the method on).

The reason I implemented it this way, rather than hard-coding the numeric 2, is that this approach is general and will work without modification for the next model version.